### DISCUSSION TITLE
How to avoid infinite loop in function definition?

### DISCUSSION THREAD

**Discussion by tristone13th** [fish-shell/fish-shell#9194] [Q&A] [answered]
if I define a function like this
```fish
function sudo
    sudo -E $argv
end
```
it will say:  The function “sudo” calls itself immediately, which would result in an infinite loop.

How to avoid this?

**Comment by tristone13th** [comment, marked as answer]
Add `command` works for me.

```fish
function sudo
    command sudo -E $argv
end
```

**Comment by faho** [comment]
See https://fishshell.com/docs/current/language.html#defining-aliases:

> If the alias has the same name as the aliased command, you need to prefix the call to the program with `command` to tell fish that the function should not call itself, but rather a command with the same name. If you forget to do so, the function would call itself until the end of time. Usually fish is smart enough to figure this out and will refrain from doing so (which is hopefully in your interest).
